tests/test_cli.py

```diff
@@ -131,6 +131,75 @@ def test_cli_data_parsed_by_question_type(
     assert answers["question"] == expected
 
 
+@pytest.mark.parametrize(
+    "type_, answer, expected",
+    [
+        ("str", "string", "string"),
+        ("str", "1", "1"),
+        ("str", "1.2", "1.2"),
+        ("str", "true", "true"),
+        ("str", "null", "null"),
+        ("str", '["string", 1, 1.2, true, null]', '["string", 1, 1.2, true, null]'),
+        ("int", "1", 1),
+        ("float", "1.2", 1.2),
+        ("bool", "true", True),
+        ("bool", "false", False),
+        ("yaml", '"string"', "string"),
+        ("yaml", "1", 1),
+        ("yaml", "1.2", 1.2),
+        ("yaml", "true", True),
+        ("yaml", "null", None),
+        ("yaml", '["string", 1, 1.2, true, null]', ["string", 1, 1.2, True, None]),
+        ("json", '"string"', "string"),
+        ("json", "1", 1),
+        ("json", "1.2", 1.2),
+        ("json", "true", True),
+        ("json", "null", None),
+        ("json", '["string", 1, 1.2, true, null]', ["string", 1, 1.2, True, None]),
+    ],
+)
+def test_data_file_parsed_by_question_type(
+    tmp_path_factory: pytest.TempPathFactory, type_, answer, expected
+) -> None:
+    src, dst, local = map(tmp_path_factory.mktemp, ("src", "dst", "local"))
+    existing_data_file = local / "data.yml"
+    with existing_data_file.open("w") as f:
+        f.write(yaml.dump({"question": answer}))
+
+    build_file_tree(
+        {
+            (src / "copier.yml"): (
+                f"""\
+                question:
+                    type: {type_}
+                """
+            ),
+            (src / "{{ _copier_conf.answers_file }}.jinja"): (
+                """\
+                # Changes here will be overwritten by Copier
+                {{ _copier_answers|to_nice_yaml }}
+                """
+            ),
+        }
+    )
+
+    run_result = CopierApp.run(
+        [
+            "copier",
+            "copy",
+            f"--data-file={existing_data_file.absolute()}",
```

Comment:
* Simply use the `build_file_tree()` helper.
* `local` is an absolute path already, no need to convert the file path to absolute anymore.
* Just a minor cosmetic and subjective thing: I'd not introduce a local variable for the data file for simplicity.

```suggestion
    build_file_tree(
        {
            (src / "copier.yml"): (
                f"""\
                question:
                    type: {type_}
                """
            ),
            (src / "{{ _copier_conf.answers_file }}.jinja"): (
                """\
                # Changes here will be overwritten by Copier
                {{ _copier_answers|to_nice_yaml }}
                """
            ),
            (local / "data.yml"): yaml.dump({"question": answer}),
        }
    )

    run_result = CopierApp.run(
        [
            "copier",
            "copy",
            f"--data-file={local / 'data.yml'}",
```